ecommerce/src/main/java/com/algolia/instantsearch/examples/ecommerce/EcommerceActivity.java

```diff
@@ -1,29 +1,48 @@
 package com.algolia.instantsearch.examples.ecommerce;
 
+import android.Manifest;
+import android.app.SearchManager;
 import android.content.Intent;
+import android.content.pm.PackageManager;
 import android.graphics.drawable.Drawable;
+import android.net.Uri;
 import android.os.Bundle;
+import android.provider.Settings;
+import android.support.annotation.NonNull;
+import android.support.design.widget.BaseTransientBottomBar;
+import android.support.design.widget.Snackbar;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.app.DialogFragment;
+import android.support.v4.content.ContextCompat;
 import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.view.Gravity;
 import android.view.View;
 import android.widget.Button;
+import android.widget.TextView;
+import android.widget.Toast;
 
 import com.algolia.instantsearch.helpers.InstantSearch;
 import com.algolia.instantsearch.helpers.Searcher;
 import com.algolia.instantsearch.ui.views.SearchBox;
 import com.squareup.leakcanary.RefWatcher;
 
+import static android.Manifest.permission.RECORD_AUDIO;
+
 public class EcommerceActivity extends AppCompatActivity {
 
     private static final String ALGOLIA_APP_ID = "latency";
     private static final String ALGOLIA_INDEX_NAME = "bestbuy_promo";
     private static final String ALGOLIA_API_KEY = "91e5b0d48d0ea9c1eb7e7e063d5c7750";
+    public static final int ID_REQ_VOICE_PERM = 1;
```

Comment:
Add PERMISSION instead of REQ

Reply:
This key reads as `ID[ENTIFIER]_REQ[UEST]_VOICE_PERM[ISSON]`. Would you rather explicit each part to `public static final int IDENTIFIER_REQUEST_VOICE_PERMISSON`?